Add support for multiple network in inspect

vdemeester opened the pull request:
To be consistent with other inspect command (on container and images),
add the possiblity to pass multiple network to the network inspect
commands. 🐼

`docker network inspect host bridge none` is possible now.

``` bash
$ docker network inspect host none
[
    {
        "name": "host",
        "id": "9030c549b1c3cf287aa7391d4419a9732e1fd629e1793d853bf0cea6869bb115",
        // […]
    },{
        "name": "none",
        "id": "7d8cf53ce05dcfaaf7a09a058a5ad4a71fe7a059c0a68bcb01e2ff32aab3a38d",
        // […]
    }
]
$ docker network inspect foobar none
Error: No such network: foobar
[
    {
        "name": "none",
        "id": "7d8cf53ce05dcfaaf7a09a058a5ad4a71fe7a059c0a68bcb01e2ff32aab3a38d",
        // […]
    }
]
$ docker network inspect foobar
Error: No such network: foobar
```
- [ ] Probably needs some docs update, on it :wink: #17161

Closes #17023 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

tiborvass commented on api/client/network.go:
```diff
@@ -184,29 +184,40 @@ func (cli *DockerCli) CmdNetworkLs(args ...string) error {
-	if err := json.NewDecoder(bytes.NewReader(obj)).Decode(networkResource); err != nil {
-		return err
+	indented := new(bytes.Buffer)
+	indented.WriteString("[\n")
```
@vdemeester sorry but whats this for again?

vdemeester replied:
To return a JSON list (the same is done for `inspect` too :wink:)

cpuguy83 commented:
Can you use json.MarshalIndent instead?

See https://github.com/docker/docker/blob/master/api/client/volume.go#L105-L173

vdemeester commented:
@cpuguy83 yes I can :smile_cat:

cpuguy83 commented:
The container inspect is complicated and weird for various reasons, but shouldn't have those issues here.

vdemeester commented:
@cpuguy83 @tiborvass Updated, it's cleaner :wink:

cpuguy83 commented:
Test failure looks legit

vdemeester commented:
@cpuguy83 yep.. weird it did not happen on my laptop.. lets see why :sweat_smile:

vdemeester committed:
Add support for multiple network in inspect

To be consistent with other inspect command (on container and images),
add the possiblity to pass multiple network to the network inspect
commands.

`docker network inspect host bridge none` is possible now.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

cpuguy83 commented:
LGTM, but does this need some doc updates?

tiborvass commented:
LGTM

thaJeztah commented:
Do we know if the completion scripts need to be modified to support this?

thaJeztah commented:
Thanks @vdemeester, think this needs changes in;
https://github.com/docker/docker/blob/master/docs/reference/commandline/network_inspect.md

(Man-page changes will be handled in https://github.com/docker/docker/issues/17161)

This is a client-only change, right? No API changes?

thaJeztah commented:
Just chatted up with @moxiegirl, and she'll take care of the docs changes in another PR. This is not regular procedure, but she's working on a rewrite of the documentation in that area, so it's easier to include this in that rewrite

so, LGTM!!

thaJeztah commented:
Failing tests on userns are not-related

thaJeztah commented:
Removed the "changeling" label; it's only to be mentioned in the changes "since 1.9-RC1"

albers commented:
Thanks @vdemeester for taking care of this!
